### PR TITLE
Generalized Stepper Implementation

### PR DESCRIPTION
**Pull request by blurpesec**
### Description

This pr introduces a Generalized Stepper that is implemented into the Send-Assets flow. It also adds a number of tests to make sure that it works correctly :open_mouth:. This will be used in DeFiZap implementation. Eventually other flows should have it incorporated. in order to reduce QA time, it's only been introduced into one flow so far (Send Assets).

### Steps to Test:
1) Add an account (metamask or another normal (non-Frame) wallet type) on https://mycryptobuilds.com/pr/2996/
2) Attempt to send a transaction using this account in the normal Send Assets flow (https://mycryptobuilds.com/pr/2996/#/send).
3) Does it work? Does it stall on one of the steps from send flow?

#### ToDo:
- [x] Implement into flow (Send Flow)
- [x] React Testing Implementation

### CODE REVIEW

File: common/v2/features/SendAssets/SendAssets.tsx [hunk 1]
```diff
@@ -1,102 +1,97 @@
...
+      component: SignTransaction,
+      props: (({ txConfig }) => ({ txConfig }))(txFactoryState),
+      actions: (payload: ITxReceipt | ISignedTx, cb: any) => handleSignedWeb3Tx(payload, cb)
     },
-    { label: '', component: SignTransaction, action: handleSignedWeb3Tx },
     {
       label: translateRaw('TRANSACTION_BROADCASTED'),
       component: TransactionReceipt,
-      action: goToDashoard
+      actions: () => goToDashoard,
```
[the remark this answers is not in the document]
Comment (blurpesec) [reply common/v2/features/SendAssets/SendAssets.tsx]: nice catch :D

File: common/v2/components/GeneralStepper/GeneralStepper.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  const getStep = (stepIndex: number) => {
+    const path = steps;
+    const { label, component, actions, props } = steps[stepIndex]; // tslint:disable-line
+    return { currentPath: path, label, Step: component, stepAction: actions, props };
+  };
+
+  const { currentPath, label, Step, stepAction } = getStep(step);
+
+  const getBackBtnLabel = () =>
+    Math.max(-1, step - 1) === -1
```
Comment (FrederikBolding) [inline common/v2/components/GeneralStepper/GeneralStepper.tsx]: Can we get a comment that explains this?

File: common/v2/components/GeneralStepper/GeneralStepper.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      onBack={goBack}
+      backBtnText={getBackBtnLabel()}
+      heading={label}
+      stepper={{ current: step + 1, total: currentPath.length }}
+    >
+      <Step
+        onComplete={(payload: any) =>
+          stepAction ? stepAction(payload, goToNextStep, goToPrevStep) : goToNextStep()
+        }
+        completeButtonText={completeBtnText || translateRaw('SEND_ASSETS_SEND_ANOTHER')}
```
Comment (FrederikBolding) [inline common/v2/components/GeneralStepper/GeneralStepper.tsx]: Is this a good default?

File: common/v2/features/SendAssets/SendAssets.tsx [hunk 4]
```diff
@@ -1,102 +1,97 @@
...
     handleSignedTx,
     handleSignedWeb3Tx,
-    txConfig: txConfigState,
-    txReceipt: txReceiptState
+    txFactoryState
   } = useStateReducer(TxConfigFactory, { txConfig: txConfigInitialState, txReceipt: null });
 
   // tslint:disable-next-line
-  const goToDashoard = () => {};
+  const goToDashboard = () => {};
```
Comment (FrederikBolding) [inline common/v2/features/SendAssets/SendAssets.tsx]: Is this intentional?

File: common/v2/components/GeneralStepper/GeneralStepper.tsx (hunk 3 above)
Comment (blurpesec) [reply common/v2/components/GeneralStepper/GeneralStepper.tsx]: It correlates to "Send Another Transaction", so I figured it'd be fine, bc stepper should by-default be used for sending transactions.

File: common/v2/components/GeneralStepper/GeneralStepper.tsx (hunk 2 above)
Comment (blurpesec) [reply common/v2/components/GeneralStepper/GeneralStepper.tsx]: will do c:

File: common/v2/features/SendAssets/SendAssets.tsx (hunk 4 above)
Comment (blurpesec) [reply common/v2/features/SendAssets/SendAssets.tsx]: yeet.

**Pull request merged**